### PR TITLE
Feature 2925 update contributions docs

### LINKED ISSUE #2925: Update contributions documentation
## ToDo
The contributions documentation needs some updates:
- Wiki -> link to CONTRIBUTING.md (in the long term, we want to make the Wiki obsolete)
- CONTRIBUTING.md:
  - Mention new `community` branch
  - Always create an issue (optional: discuss with SecHub core team) before contributing something

### PR DESCRIPTION
**Pull request by sven-dmlr**
- merged wiki information into CONTRIBUTING.md
- updated top level readme.adoc
- improved docs for documentation
- closes #2925

### CODE REVIEW

File: README.adoc [hunk 1]
```diff
@@ -45,6 +46,9 @@ Please visit https://github.com/mercedes-benz/sechub/wiki/ for detailed informat
 
 === Introduction
 
+SecHub orchestrates various security and vulnerability scanners which can find potential vulnerabilities in the software under test.
```
Comment (haerter-tss) [inline README.adoc:49]:
```suggestion
SecHub orchestrates various security and vulnerability scanners which can find potential vulnerabilities in sourcecode, binaries or web applications.
```

File: README.adoc [hunk 2]
```diff
@@ -53,27 +57,29 @@ Please visit https://github.com/mercedes-benz/sechub/wiki/ for detailed informat
...
                                                    +--------------+
 ....
 
-SecHub orchestrates various security and vulnerability scanners which can find potential vulnerabilities in the software under test (see: <<figure-architecture-overview, Architecture overview>>). This enables security, development and operation teams to review and fix security issues. As a result, SecHub improves application security.
-
 The objective of SecHub is to help secure the software development lifecyle (SDLC) phases: development, deployment and maintenance. From the first written line of code to the application being in production. SecHub can be used to scan the software continuously.
 
-To reduce the cognitive load for the user, the security tools are categorized into modules. Modules are named after the security testing method they perform: `codeScan`, `webScan`, `infraScan` etc.
+To reduce the cognitive load for the user, the security tools are categorized into modules.
+Modules are named after the security testing method they perform: `codeScan`, `licenseScan`, `secretScan`, `webScan` etc.
```
Comment (haerter-tss) [inline README.adoc:71]:
```suggestion
The security tools are categorized into modules which are named after the security testing method they perform: `codeScan`, `licenseScan`, `secretScan`, `webScan` etc.
```

File: README.adoc [hunk 3]
```diff
@@ -53,27 +57,29 @@ Please visit https://github.com/mercedes-benz/sechub/wiki/ for detailed informat
...
+--
 
 Because of the modules, the user only needs to understand what security testing a module performs, rather than to know what specific security tools are used for the actual scan. The user describes in a configuration file what module(s) to use. Or in other words, what security testing methods should be used to test the software with.
 
-To start a scan, the user sends the configuration file to the SecHub server via API. Even better, SecHub provides a client which calls the REST API for the user. The SecHub client can be used manually or can be integrated into a continues-integration build pipeline (see: <<figure-cicd-user-sechub, Scan using CI/CD or User>>).
+To start a scan, the user sends the configuration file to the SecHub server via API. Even better, SecHub provides a client which calls the REST API for the user. The SecHub client can be used manually or can be integrated into a continues-integration build pipeline (see below figure).
```
Comment (haerter-tss) [inline README.adoc:80]:
```suggestion
To start a scan, the user sends the configuration file to the SecHub server via API. For ease of use SecHub offers a client which calls the REST API for the user. The SecHub client can be used manually or can be integrated into a continues-integration build pipeline (see below figure).
```

File: README.adoc [hunk 4]
```diff
@@ -250,12 +255,16 @@ SecHub collects the scan results from various security tools and converts them i
...
+The report can be downloaded in two flavors: JSON and HTML. Both are human readable. The HTML report is self-contained and can be read in any browser.
+The JSON format is machine readable and can be read by the <<sechub-plugins, SecHub plugins>>.
 
 [[mark-false-positives]]
 ==== Mark Findings as False-Positives
 
-There are two major reasons for marking a security finding as false-positive. First, it is an actual false-positive. Second, the finding is a false-positive in the context of the application. For example, the application is never deployed to be reachable from the internet.
+There are two major reasons for marking a security finding as false-positive. +
+- First, it is an actual false-positive. +
+- Second, the finding is a false-positive in the context of the application. For example, the application is never deployed to be reachable from the internet.
```
Comment (haerter-tss) [inline README.adoc:266]:
```suggestion
There are two major reasons for marking a security finding as false-positive: +
- It is an actual false-positive. +
- The finding is a false-positive in the context of the application. For example, the application is never deployed to be reachable from the internet.
```

File: README.adoc [hunk 5]
```diff
@@ -279,10 +288,10 @@ Security tools are categorized into modules.
 Each module performs a different security testing method:
 
 * <<codescan, `codeScan`>> - scans the code for potential vulnerabilities (weaknesses). Also known as SAST or static source code analysis.
```
Comment (haerter-tss) [inline README.adoc:290]:
```suggestion
* <<codescan, `codeScan`>> - scans the code or binaries for potential vulnerabilities (weaknesses). This includes SAST (static application security testing) and IaC (infrastructure as code).
```

File: README.adoc [hunk 6]
```diff
@@ -335,43 +352,37 @@ More details: https://mercedes-benz.github.io/sechub/latest/sechub-client.html#i
...
 
 For more details about the architecture have a look at the architecture documentation: https://mercedes-benz.github.io/sechub/latest/sechub-architecture.html.
 
 === Operations Perspective
 
-The operations team has to do most of the work. The operations team of SecHub needs to configure the scan tools as well as manage users and projects. This is an intentional design choice. The idea is, that users like security experts, operations engineers and developers can just scan and therefore focus on their work, rather than take care of SecHub and security scanners.
+The operations team has to do most of the work.
+The operations team of SecHub needs to configure the scan tools as well as manage users and projects.
+This is an intentional design choice.
+The idea is, that users like security experts, operations engineers and developers can just scan and therefore focus on their work, rather than take care of SecHub and security scanners.
```
Comment (haerter-tss) [inline README.adoc:379]: Maybe we should cut this part out. Users might not split their teams in DEV and OPS and instead have the same people run and use SecHub.

File: sechub-doc/src/docs/asciidoc/documents/techdoc/04_documentation.adoc [hunk 7]
```diff
@@ -7,28 +7,15 @@
...
-
-Having now moved to `github.com` we can write those files of course
-also as `.adoc`
-====
-
 
 [[section-documentation-asciidoc]]
 ==== Asciidoc
-When ever writing a full scaled documentation - means more than some lines to explain a folder etc. - we do this with Asciidoc.
+When ever writing a full scaled documentation - we do this with Asciidoc. +
```
Comment (haerter-tss) [inline sechub-doc/src/docs/asciidoc/documents/techdoc/04_documentation.adoc:17]:
```suggestion
When a full scaled documentation is needed it should be done with Asciidoc. +
```

File: sechub-doc/src/docs/asciidoc/documents/techdoc/04_documentation.adoc [hunk 8]
```diff
@@ -7,28 +7,15 @@
 [[section-documentation-markdown]]
 ==== Markdown
 Simple technical information -e.g. about the content of an folder
-- will be written in markdown syntax.
-
-Normaly we use a README.md file to explain simple things.
+- will be written in a README.md with markdown syntax.
```
Comment (haerter-tss) [inline sechub-doc/src/docs/asciidoc/documents/techdoc/04_documentation.adoc:10]:
```suggestion
- should be written in a README.md with markdown syntax.
```

File: sechub-doc/src/docs/asciidoc/documents/techdoc/04_documentation.adoc [hunk 9]
```diff
@@ -73,10 +55,9 @@ TIP: For an example look into `@IsReceivingAsyncMessage` and look also for its
 
 [[section-documentation-usecase-event-overview]]
 ===== Usecase event overview diagrams
-For some usecases we got event trace information (from special integration tests). With those information
+For some usecases we got event trace information (from special integration tests). With those informations
```
Comment (haerter-tss) [inline sechub-doc/src/docs/asciidoc/documents/techdoc/04_documentation.adoc:58]:
```suggestion
For some usecases we got event trace information (from special integration tests). With this information
```

File: sechub-doc/src/docs/asciidoc/documents/techdoc/04_documentation.adoc [hunk 10]
```diff
@@ -73,10 +55,9 @@ TIP: For an example look into `@IsReceivingAsyncMessage` and look also for its
...
 additional diagrams are generated and included at Usecase description level. They show the events happening
```
Comment (haerter-tss) [inline sechub-doc/src/docs/asciidoc/documents/techdoc/04_documentation.adoc:59]:
```suggestion
additional diagrams are generated and included at usecase description level. They show the events happening
```

**Review by haerter-tss — APPROVED** [review]
If you like you can add my suggestions before merging.

File: CONTRIBUTING.md [hunk 11]
```diff
@@ -1,40 +1,102 @@
...
+Example:
+```
+Described commit message format #868
+
+- added examples
+- explained purpose
+```
+
+### Development tools setup
+Of course it depends on you which kind of IDE you use for development. But we have some defaults defined - e.g. for git configuration which should be used.
```
[the remark this answers is not in the document]
Comment (sven-dmlr) [reply CONTRIBUTING.md:94]: changed

File: README.adoc (hunk 1 above)
Comment (sven-dmlr) [reply README.adoc:49]: changed

File: README.adoc (hunk 2 above)
Comment (sven-dmlr) [reply README.adoc:71]: changed

File: README.adoc (hunk 3 above)
Comment (sven-dmlr) [reply README.adoc:80]: changed

File: README.adoc (hunk 4 above)
Comment (sven-dmlr) [reply README.adoc:266]: changed

File: README.adoc (hunk 5 above)
Comment (sven-dmlr) [reply README.adoc:290]: changed

File: README.adoc (hunk 6 above)
Comment (sven-dmlr) [reply README.adoc:379]: rewrote the section

File: sechub-doc/src/docs/asciidoc/documents/techdoc/04_documentation.adoc (hunk 8 above)
Comment (sven-dmlr) [reply sechub-doc/src/docs/asciidoc/documents/techdoc/04_documentation.adoc:10]: changed

**Pull request merged**